alvr/launcher/src/actions.rs

```diff
@@ -0,0 +1,348 @@
+                    progress: 0.5,
+                }))
+                .unwrap(),
+        }
+    }
+
+    Ok(buffer)
+}
+
+async fn install(
+    tx: &Sender<WorkerMessage>,
+    release: ReleaseInfo,
+    client: &reqwest::Client,
+) -> anyhow::Result<()> {
+    tx.send(WorkerMessage::ProgressUpdate(Progress {
+        msg: "Starting install".to_string(),
+        progress: 0.0,
+    }))
+    .unwrap();
+
+    let file_name = if cfg!(windows) {
+        "alvr_streamer_windows.zip"
+    } else {
+        "alvr_streamer_linux.tar.gz"
+    };
+
+    let url = release
+        .assets
+        .get(file_name)
+        .ok_or(anyhow::anyhow!("Unable to determine download link"))?;
+
+    let buffer = download(tx, "Downloading Streamer", url, client).await?;
+
+    let installation_dir = installations_dir().join(&release.tag);
+
+    fs::create_dir_all(&installation_dir)?;
+
+    let mut buffer = Cursor::new(buffer);
+    if cfg!(windows) {
+        zip::ZipArchive::new(&mut buffer)?.extract(&installation_dir)?;
+    } else {
+        tar::Archive::new(&mut GzDecoder::new(&mut buffer)).unpack(&installation_dir)?;
+    }
+
+    Ok(())
+}
+
+pub fn data_dir() -> PathBuf {
+    if cfg!(target_os = "linux") {
+        PathBuf::from(env::var("HOME").expect("Failed to determine home directory"))
+            .join(".local/share/ALVR-Launcher")
+    } else {
+        env::current_dir()
+            .expect("Unable to determine executable directory")
+            .join("ALVR-Launcher")
+    }
+}
+
+pub fn get_installations() -> Vec<InstallationInfo> {
+    match fs::read_dir(installations_dir()) {
+        Ok(entries) => entries
+            .into_iter()
+            .filter_map(|entry| {
+                entry
+                    .ok()
+                    .filter(|entry| match entry.file_type() {
+                        Ok(file_type) => file_type.is_dir(),
+                        Err(why) => {
+                            eprintln!("Failed to read entry file type: {}", why);
+                            false
+                        }
+                    })
+                    .map(|entry| {
+                        let mut apk_path = entry.path();
+                        apk_path.push(APK_NAME);
+                        InstallationInfo {
+                            version: entry.file_name().to_string_lossy().to_string(),
+                            is_apk_downloaded: apk_path.exists(),
+                        }
+                    })
+            })
+            .collect(),
+        Err(why) => {
+            eprintln!("Failed to read versions dir: {}", why);
+            Vec::new()
+        }
+    }
+}
+
+pub fn launch_dashboard(version: &str) -> Result<()> {
+    let installation_dir = installations_dir().join(version);
+
+    let dashboard_path = if cfg!(windows) {
+        installation_dir.join("ALVR Dashboard.exe")
+    } else if cfg!(target_os = "linux") {
+        installation_dir.join("bin/alvr_alvr_dashboard")
```

Comment:
an error :)

Reply:
Thank, I'll fix that

Reply:
checked path, it should be `alvr_streamer_linux/bin/alvr_dashboard` here